### PR TITLE
Add LDAP Server

### PR DESCRIPTION
**Pull request by mhuguesaws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: 1.architectures/6.ldap_server/cf_ldap_server.yaml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    Type: List<AWS::EC2::SecurityGroup::Id>
+    Description: 'Security Group Ids'
+  Password:
+    Type: String
+    Description: 'LdapServer'
+    NoEcho: true
+############################
+## AWS Batch Infrastructure
+Resources:
+  LdapKeyPair:
```
Comment (sean-smith) [inline 1.architectures/6.ldap_server/cf_ldap_server.yaml:38]: Does the user need this to access to the LDAP instance?

File: 1.architectures/6.ldap_server/cf_ldap_server.yaml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  SubnetId:
+    Type: AWS::EC2::Subnet::Id
+    Description: 'Subnet Id'
+  SecurityGroupId:
+    Type: List<AWS::EC2::SecurityGroup::Id>
+    Description: 'Security Group Ids'
+  Password:
+    Type: String
+    Description: 'LdapServer'
+    NoEcho: true
```
Comment (sean-smith) [inline 1.architectures/6.ldap_server/cf_ldap_server.yaml:34]: Can this be avoided by using Secrets Manager?

File: 1.architectures/6.ldap_server/cf_ldap_server.yaml (hunk 1 above)
Comment (mhuguesaws) [reply 1.architectures/6.ldap_server/cf_ldap_server.yaml:38]: A User won't need access.
An admin will want to access the instance for various reasons: maintenance, upgrades etc...

File: 1.architectures/6.ldap_server/cf_ldap_server.yaml (hunk 2 above)
Comment (mhuguesaws) [reply 1.architectures/6.ldap_server/cf_ldap_server.yaml:34]: We can use secret manager to generate a secure string. It won't handle password rotation.

**Review by nghtm — APPROVED** [review]
Looks good - adding a ReadMe is my only feedback. 

Tested the template in my isengard account, deployed successfully in a public subnet, but do not know what to do next without any setup instructions

**Comment by mhuguesaws** [comment]
Let me add some documentation before merging. Forgot about that.

**Comment by sean-smith** [comment]
> Looks good - adding a ReadMe is my only feedback.
> 
> Tested the template in my isengard account, deployed successfully in a public subnet, but do not know what to do next without any setup instructions

I think we should include the setup instructions for both PC and SMHP as a separate PR.

File: 1.architectures/6.ldap_server/cf_ldap_server.yaml [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    AllowedValues:
+      - t3a.micro
+      - t3a.small
+      - t3a.medium
+      - t3a.large
+      - t2.small
+  LatestUbuntuAmiId:
+    Type: AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>
+    Default: '/aws/service/canonical/ubuntu/server/22.04/stable/current/amd64/hvm/ebs-gp2/ami-id'
+    Description: 'Ubuntu 22.04 AMI Id'
```
Comment (perifaws) [inline 1.architectures/6.ldap_server/cf_ldap_server.yaml:20]: `Id` -> `ID`

File: 1.architectures/6.ldap_server/cf_ldap_server.yaml [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+
+
```
Comment (perifaws) [inline 1.architectures/6.ldap_server/cf_ldap_server.yaml:82]: remove extra line

File: 1.architectures/6.ldap_server/cf_ldap_server.yaml [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+  LdapInstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    DependsOn: LdapSecretPolicy
+    Properties:
+      Roles:
+        - !Ref LdapInstanceRole
+
+
```
Comment (perifaws) [inline 1.architectures/6.ldap_server/cf_ldap_server.yaml:90]: remove extra line

**Review by perifaws — CHANGES_REQUESTED** [review]

**Comment by sean-smith** [comment]
What's needed here?

**Comment by mhuguesaws** [comment]
more documentation. It is coming.

**Comment by mhuguesaws** [comment]
@sean-smith minimal doc completed. Ready for your review.

**Review by sean-smith — APPROVED** [review]

**Review by perifaws — APPROVED** [review]

**Pull request merged**